eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -401,6 +384,40 @@ steps:
     dropMetadataContainerName: 'DropMetadata-ProfilingInputs'
   condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
 
+- task: MicroBuildBuildVSBootstrapper@2
+  displayName: 'Build a Visual Studio bootstrapper for tests'
+  inputs:
+    channelName: "$(VsTargetChannelForTests)"
+    vsMajorVersion: "$(VsTargetMajorVersion)"
+    manifests: '$(Build.Repository.LocalPath)\artifacts\VS15\Microsoft.VisualStudio.NuGet.Core.vsman'
+    outputFolder: '$(Build.Repository.LocalPath)\artifacts\VS15'
+  condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'))"
+
+- task: PublishBuildArtifacts@1
```

Comment:
Given that these artifacts are created in the same directory, do we have any overlap concerns? Or does the above task create the json file with a different name.

Reply:
Great question Nikolche. I wanted to address this in the PR description but missed it. In the PR build and official build, I triggered for this change, I noticed that a single `bootstrapper.json` file got generated with two entries. First entry pointing to `bootstrapper for optprof` because it was generated first in the pipeline and the second entry pointing to `bootstrapper for tests` because it was generated later in the pipeline.

For example, private build for this PR produced following `bootstrapper.json` file

```json
[
    {
        "overlaidChannelManifestUrl":  "REMOVED",
        "VSBranch":  "main",
        "VSManifestName":  "VisualStudio.17.int.main",
        "VSBuildVersion":  "17.5.33031.37",
        "BuildDrop":  "REMOVED/0110c214-cfdb-b409-3101-6609d74dc856",
        "InputManifests":  "D:\\a\\_work\\1\\s\\artifacts\\VS15\\Microsoft.VisualStudio.NuGet.Core.vsman",
        "bootstrapperUrl":  "REMOVED/fc5794c1-45c5-487f-9254-27215b06a90e/vs_enterprise.exe"
    },
    {
        "overlaidChannelManifestUrl":  "REMOVED",
        "VSBranch":  "d17.4",
        "VSManifestName":  "VisualStudio.17.int.d17.4",
        "VSBuildVersion":  "17.4.33027.239",
        "BuildDrop":  "REMOVED/1b6eebc8-6b11-52dd-106f-345caa00c389",
        "InputManifests":  "D:\\a\\_work\\1\\s\\artifacts\\VS15\\Microsoft.VisualStudio.NuGet.Core.vsman",
        "bootstrapperUrl":  "REMOVED/c9353840-ed32-4749-abf9-b129739292ba/vs_enterprise.exe"
    }
]
```

Another notable difference here is that the GUID in `bootstrapperUrl` in the above json file is different for `int.main` channel and `.int.d17.4` channel.

The artifacts uploaded to the build pipeline also confirms the same.

<img width="305" alt="image" src="https://user-images.githubusercontent.com/52756182/199114953-3207c221-26f8-4319-afeb-abaad64b419e.png">

The subfolder with guid `c9353840-ed32-4749-abf9-b129739292ba` is used for apex/e2e tests and folder name whereas the subfolder with guid `fc5794c1-45c5-487f-9254-27215b06a90e` is used for optprof pipeline.